### ISSUE TITLE
Use Record<string, never> instead of {} for empty types

### ISSUE THREAD

**Issue by tobiasdiez** [dotansimha/graphql-code-generator#6068]
**Is your feature request related to a problem? Please describe.**
By default the typescript resolver plugin creates typing information of the form
```ts
export type Resolver<TResult, TParent = {}, TContext = {}, TArgs = {}> = ...

export type ResolversParentTypes = ResolversObject<{
  Mutation: {};
  Query: {};
}
```
The expectation here is that `{}` denotes the empty type (which only accepts the empty variable `{}`). However, this is not the case:
```ts
interface AAA {
  aaa: {};
}

const x: AAA = { aaa: true };
```
compiles just fine. Indeed, the type `{}` stands for any non-nullish value. For more details and examples see https://github.com/typescript-eslint/typescript-eslint/issues/2063#issuecomment-675156492.

**Describe the solution you'd like**
Using `Record<string, never>` instead of `{}` at least for parents and args (not sure about the default context).

**Describe alternatives you've considered**

**Additional context**

**Comment by Urigo** [comment]
Hi @tobiasdiez and thank you for the report!

Sorry but I'm not adding a lot here but just labeling it according to our new Contribution Guide and [issue flow](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md#a-typical-contributor-workflow).

It seems already got into `stage 0`.

Now in order to advance to `stage 1` we'll need a simple reproduction, maybe in code-sandbox? 

Later to progress to `stage 2`, a failing test would be needed, would be great if someone could help progress the issues through the [stages](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md#a-typical-contributor-workflow).

Thank you and sorry that this comment is not a complete solution (yet).

**Comment by fivethreeo** [comment]
Is stage 1 needed if you can make a failing test? I think a test would show this better. This should be a test that is successful when type checking fails.

**Comment by Urigo** [comment]
@fivethreeo if you can create a failing test, can can move directly to stage 2

**Comment by fivethreeo** [comment]
If i understand correctly this test should do, but I cannot make it fail with Record<string, never>

**Comment by fivethreeo** [comment]
@Urigo made a test for this now: https://github.com/dotansimha/graphql-code-generator/pull/6697

`{}` seems to work just like ` Record<never, never>`, not sure if it is more correct to use  Record.

**Comment by tobiasdiez** [comment]
While I agree that the code is still running, you do loose  quite a bit of typing safety. Can this please be reconsidered?

With 
```ts
export type ResolverFn<TResult, TParent, TContext, TArgs> = (
  parent: TParent,
  args: TArgs,
  context: TContext,
  info: string
) => Promise<TResult> | TResult;
export type Resolver<TResult, TParent = {}, TContext = {}, TArgs = {}> = ResolverFn<TResult, TParent, TContext, TArgs>
```
and a resolver
```ts
const me: Resolver<string> = function() {
    return "test"
}
```
The following code works just fine without any typescript errors:
```ts
me("testParent", "testArgs", {}, "testInfo")
```

However, if you do change the default types in the definition of a resolver to `Record<string, never>`, then typescript complains that `Argument of type 'string' is not assignable to parameter of type 'Record<string, never>'`, which is what you expect.

Please have a look at this ts playground illustrating the problem (with condensed code as generated by the graphql generator): https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBDAnmYcBKwDOEA2A3YKAMQDsAeAFQ0wFccYAaOCgBQEMpgTHmBhCbqB4UAglADmmAHxwAvHAAUAKDhwwHLjABczdp24MVcDpJ2iJmQ6oDGAmELP9BIRkYCWJAGYQdmGFA9xJQBKORkWKAgAWzdMYEpqOhgZAB9mRPoAbiUc0EhYBGRUalwCKASsJKZWDW45OABvAF9qp3sXeubqsUl6jFsoABMyPwCScSYSYDKpGXkS-EJSCtp6ar1NVrshbospXPBoeCQUdCxSwkwN7goizE6jVQBFGkJEHWbspoP846Kz7CLKCYW4oe7yBqPVRwACqcSgvn8gS+PyOcA89ignjY1lQbSEjW+SjyaJOqBebwWZUwZHxLlBqHkdJ41xgDLgQi4g3uVMurIZmAA2gByClQRDCgC6fXOQKutTZdxFYolkrmjSMUWAAH4dLzyvqQUrhXDCFKmPyikxmQypF9skoyXAALLAfX1FWGkVaqVKAD0fuhAD1tTlbCQ-HAtTpXe75J4aCRrDA3AIFKFIdC4JwYDQoCQ4AAiex+QtKIkJpMpgRwbHJ6CIBScQFlGNu2VlDPfOswBsKLXBHJahTFrAwVmFpijvw9TCTwlTkswACSXgghcHQA

**Comment by charlypoly** [comment]
@tobiasdiez 

You are right, actually, an empty interface should never be used, as stated in the TypeScript FAQ: https://github.com/microsoft/TypeScript/wiki/FAQ#why-are-all-types-assignable-to-empty-interfaces

Would you have time to do a PR to fix this? 👀

**Comment by eddeee888** [comment]
This should be fixed in `typescript-resolvers` v5: https://github.com/dotansimha/graphql-code-generator/blob/master/packages/plugins/typescript/resolvers/CHANGELOG.md#500

THanks all!
